Sonal/codebuilder beta (#4007)
forcedotcom/salesforcedx-vscode: docs/_articles/en/vscode-desktop/recommended-extensions.md

---
title: Other Recommended Extensions
lang: en
---

#  Extensions We Recommend
In addition to the extension pack shipped by Salesforce, the following extensions are useful for Salesforce developers. We also package the Extension pack with some of the most-used community extensions in the [expanded extension pack](https://marketplace.visualstudio.com/items?itemName=salesforce.salesforcedx-vscode-expanded).

> > **_NOTE:_** Because Code Builder is a web-based and comes with pre-installed extensions, this list is not of relevance if Code Builder is your development environment of choice.

Reviewer: This explanation is wordy. Can you pare it down? Is this a beta limitation or the way it'll always be for CB? 
I would word this differently depending on if we plan to allow other extensions in the future. If not:
You can install more extensions only if using Salesforce Extensions for VS Code on your desktop. Code Builder comes fully configured so you can't install additional software (or just extensions).

> ## Apex PMD

Reviewer: PMD ??

Reviewer: Or is that its literal name? heh

Allows you to run Apex static analysis directly in VS Code on Apex and Visualforce files.

<a class="slds-button slds-button_brand landing__header-cta slds-m-vertical--x-large" href="https://marketplace.visualstudio.com/items?itemName=chuckjonas.apex-pmd">Install Apex PMD</a>

## Apex Log Analyzer

> An analyzer for Salesforce debug logs aimed at making performance analysis much easier and quicker. You may also find it generally useful for quickly understanding how your code is executing.

Reviewer: ```suggestion
An analyzer for Salesforce debug logs aimed at making performance analysis much easier and quicker. It's also useful for quickly understanding how your code is executing.
```
